contracts/strategy/yearn/YearnStrategy.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+
+import {PercentMath} from "../../lib/PercentMath.sol";
+import {ERC165Query} from "../../lib/ERC165Query.sol";
+import {IStrategy} from "../IStrategy.sol";
+import {CustomErrors} from "../../interfaces/CustomErrors.sol";
+import {IYearnVault} from "../../interfaces/yearn/IYearnVault.sol";
+import {IVault} from "../../vault/IVault.sol";
+
+contract YearnStrategy is IStrategy, AccessControl, CustomErrors {
+    using SafeERC20 for IERC20;
+    using PercentMath for uint256;
+    using ERC165Query for address;
+
+    error StrategyNoUnderlying();
+    error StrategyNotEnoughShares();
+
+    bytes32 public constant MANAGER_ROLE =
+        0x241ecf16d79d0f8dbfb92cbc07fe17840425976cf0667f022fe9877caa831b08; // keccak256("MANAGER_ROLE");
+
+    IERC20 public immutable underlying;
+    /// @inheritdoc IStrategy
+    address public immutable override(IStrategy) vault;
+    // yearn vault that this strategy is interacting with
+    IYearnVault public immutable yVault;
+
+    constructor(
+        address _vault,
+        address _owner,
+        address _yVault,
+        address _underlying
+    ) {
+        if (_owner == address(0)) revert StrategyOwnerCannotBe0Address();
+        if (!_vault.doesContractImplementInterface(type(IVault).interfaceId))
+            revert StrategyNotIVault();
```

Comment:
I think we are missing the same check for `IYearnVault` interface, and probably `_underlying`?

Reply:
hummm, I'm not sure that we need to be honest, what do you think @naps62

Reply:
That depends on the vault I guess. I don't think all (if any?) of yearn's vaults implement ERC165. At least the non-ERC4626 ones

Reply:
as for underlying, I'm assuming it's the same underlying the vault already knows right? It only needs to be checked once, not in both places.